cmd/pbstatuser/main.go

```diff
@@ -20,25 +26,179 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// This needs to be done in 4 goroutines.
+// 1) Reads messages, calls sources to get auth type and ARN and then enqueue the work
+// in one of three channels (AWS, Azure, GCP)
+// 2) Reads from AWS channel, performs availability check, has configurable throttling.
+// 3) Reads from Azure channel, performs availability check, has configurable throttling.
+// 4) Reads from GCP channel, performs availability check, has configurable throttling.
+
+// Beware there is no database connection, no job queue, most of what api/worker processes have
+// is not available because it should not be needed. The worker will simply use SDKs to check
+// statuses and send out results via kafka.Send function to Sources.
+// Also make sure that all goroutines are closed correctly on context cancel. A WaitGroup
+// would be a really good here: https://gist.github.com/fracasula/b579d52daf15426e58aa133d0340ccb0
+
+const ChannelBuffer = 1024
+
+type StatusType string
+
+const (
+	StatusUnavailable StatusType = "unavailable"
+	StatusAvaliable   StatusType = "available"
+)
+
+type Pipeline struct {
+	// Resource type of the source
+	ResourceType string `json:"resource_type"`
```

Comment:
I don't see the need of passing ResourceType along goroutines, you can simply make a local variable in the send function I guess. Or I am not sure why you have it in the struct.

Reply:
At first, I wanted to create two structs: 
one for passing information between the go routines and the second as the payload to send over to sources according to their needs: https://consoledot.pages.redhat.com/docs/dev/services/sources/integration.html#_prepare_an_endpoint_for_the_availability_checking
```
{
  "resource_type": "Application",
  "resource_id": "12345",
  "status": "unavailable",
  "error": "something went wrong with..."
}
```
Do you think I should separate the pipeline struct into two?

Reply:
Yeah I think these are two definitely different structures, at the end of the pipeline you just need to copy few values from one to other, but that is a small tax to pay for better readability. In this version by looking at the type, I am unable to tell what its main purpose.